### PR TITLE
feature: DPlayer support

### PR DESCRIPTION
**Pull request by Candinya**
## Summary

New Feature: Add [DPlayer](https://github.com/DIYgod/DPlayer) support

Videos can be played inline now, rather than opening a new window.

Related issue: #7201 

## Addition

1. ~Player language may differs (media-video.vue, Line 63, I forgot to change 😶 )~ Fixed
2. ~DPlayer full web mode would cause page style break~ ~Relieved by removing the classname immediately~ Resolved using modified version of DPlayer.

### CODE REVIEW

File: src/client/components/media-video.vue [hunk 1]
```diff
@@ -50,6 +50,29 @@ export default defineComponent({
 	created() {
 		this.hide = (this.$store.state.nsfw === 'force') ? true : this.video.isSensitive && (this.$store.state.nsfw !== 'ignore');
 	},
+	methods: {
+		onPlay() {
+			if (this.dp === null) {
+				// Only for creating new
+				this.dp = new DPlayer({
+					// Configure parameters
+					container: document.getElementById(this.video.id),
```
Comment (rinsuki) [inline src/client/components/media-video.vue]: If multiple same video links exists in one page (e.g. renote), this conflicts with other same video link.

**Review by rinsuki — CHANGES_REQUESTED** [review]

File: src/client/components/media-video.vue (hunk 1 above)
Comment (Candinya) [reply src/client/components/media-video.vue]: Reproduced. Figuring out how to resolve it. 😶

**Comment by defango** [comment]
You may wanna look into video.js we added that and its working pretty good. https://github.com/gitmolife/gitmo/commit/02e68e896247f59a1e24da3660a56b0866c51ebf

**Comment by rinsuki** [comment]
and I think DPlayer's danmaku feature _possibly_ violating patents of DWANGO Co., Ltd., but I'm not familiar with patents and related things, so I would like to hear from other collaborators/contributors.

File: src/client/components/media-video.vue (hunk 1 above)
Comment (Candinya) [reply src/client/components/media-video.vue]: Should be fixed now. Passing `$event` into the trigger function and use `event.target` to get the clicked element. 😃

**Comment by Candinya** [comment]
~Here's one more question: DPlayer full-web (not full-screen) mode would attach a class `.dplayer-web-fullscreen-fix` containing `position: fixed;` style to body element, which will break the page style.
Figuring out how to prevent this.
Using additional code to remove the class attached when received event now, while not working that perfectly.~
Resolved using modified version of DPlayer.

**Comment by Candinya** [comment]
> and I think DPlayer's danmaku feature _possibly_ violating patents of DWANGO Co., Ltd., but I'm not familiar with patents and related things, so I would like to hear from other collaborators/contributors.

I found that danmaku has been included as a W3C proposal, so there might not be legal problems. Maybe check [w3c/danmaku](https://github.com/w3c/danmaku) for more information.

**Comment by acid-chicken** [comment]
It seems they haven't reached an opinion about this yet: w3c/danmaku#36
In my opinion, the patent aims at reducing to display overlapping of comments but W3C's approach seems quite similar to it, so (unfortunately) this player may violate their patents in Japan....

**Comment by Candinya** [comment]
> It seems they haven't reached an opinion about this yet: [w3c/danmaku#36](https://github.com/w3c/danmaku/issues/36)
> In my opinion, the patent aims at reducing to display overlapping of comments but W3C's approach seems quite similar to it, so (unfortunately) this player may violate their patents in Japan....

Oops, so I'd close this PR to prevent further possible problems.
Thank you all for your participants and sincere help! 😃

**Comment by acid-chicken** [comment]
Looking into this patent in more detail, I found an interesting document. According to this, the patent appears some of the claims lost their validity in 2017.
Perhaps we can avoid violating this patent by disabling following features?:
- float displayed comments
- changing text sizes per comments
- multiline comments

You can find this document by clicking "Invalidity 2017-800123" in page <https://www.j-platpat.inpit.go.jp/c1800/PU/JP-2006-333851/7294651F33633E1EBF3DEC66FAE0ECAD878D19E1829C378FC81D26BBD0A4263B/10/en>.

**Comment by Candinya** [comment]
> Looking into this patent in more detail, I found an interesting document. According to this, the patent appears some of the claims lost their validity in ~2017~ December 2020.
> Perhaps we can avoid violating this patent by disabling following features?:
> 
> * float displayed comments
> * changing text sizes per comments
> * multiline comments
> 
> You can find this document by clicking "Invalidity 2017-800123" in page https://www.j-platpat.inpit.go.jp/c1800/PU/JP-2006-333851/7294651F33633E1EBF3DEC66FAE0ECAD878D19E1829C378FC81D26BBD0A4263B/10/en.

Yes, it's currently set as an simple player without danmaku mode ( due to some compability issues ), but if other days someone figured it out and enabled its danmaku functions, legal issues maybe occur then. 
So to prevent this possible issue, I've closed this PR and willing to try implementing another pure video player ( like video.js, @defango mentioned above )
Thanks for your sincere help 😉

**Comment by acid-chicken** [comment]
> > According to this, the patent appears some of the claims lost their validity in 2017 December 2020.

Note: this (my deleted comment) is mistaken reading the docs. It disables the claims of disabling patents, not disabling patents. I'm sorry about that jumping to wrong conclusion.

**Pull request closed without merging**